sync_events.py

```diff
@@ -0,0 +1,249 @@
+
+
+def load_events():
+    events = {}
+    for path in [Path(p) for p in glob.glob("content/en/events/*/*.md")]:
+        event = Event.load(path)
+        if event is None:
+            # not a pretix event
+            continue
+        events[event.pretix_slug] = event
+
+    return events
+
+
+if __name__ == "__main__":
+    # load events from file tree
+    events = load_events()
+
+    with open("PRETIX_API_TOKEN") as f:
+        token = f.readline()
+
+    # events from pretix
+    response = requests.get(
+        "https://pretix.eu/api/v1/organizers/correlaid/events/"
+        "?is_future=true&has_subevents=false&is_public=true&live=true",
+        headers={"Authorization": f"Token {token}"},
+    )
+    response.raise_for_status()
+
+    response_json = response.json()
+    if response_json["next"] is not None:
+        raise Exception("pagination not implemented")
+
+    api_events = {
+        api_event["slug"]: api_event for api_event in response_json["results"]
+    }
+
+    # what does that do?
+    # events new on pretix
+    for pretix_slug in set(api_events) - set(events):
+        Event.create(api_events[pretix_slug])
+    # events already on pretix and website
+    for pretix_slug in set(api_events).intersection(set(events)):
+        events[pretix_slug].update(api_events[pretix_slug])
+    # events deleted on pretix -> delete from website
+    for pretix_slug in set(events) - set(api_events):
+        events[pretix_slug].delete()
+
+    # subevents
+    response = requests.get(
+        "https://pretix.eu/api/v1/organizers/correlaid/events/"
+        "?has_subevents=true&is_public=true&live=true",
+        headers={"Authorization": f"Token {token}"},
+    )
+    response.raise_for_status()
+
+    response_json = response.json()
+    if response_json["next"] is not None:
+        raise Exception("pagination not implemented")
+
+    for slug in [
+        api_event["slug"]
+        for api_event in response_json["results"]
+        if api_event["has_subevents"]
+    ]:
+        # get subevents
+        response = requests.get(
+            f"https://pretix.eu/api/v1/organizers/correlaid/events/{slug}/subevents/"
+            "?is_future=true&active=true",
+            headers={"Authorization": f"Token {token}"},
+        )
+        response.raise_for_status()
+
+        response_json = response.json()
+        if response_json["next"] is not None:
+            raise Exception("pagination not implemented")
+
+        api_events = {
+            f"{api_event['event']}/{api_event['id']}": api_event
+            for api_event in response_json["results"]
+        }
+        for pretix_slug in set(api_events) - set(events):
+            Event.create(api_events[pretix_slug])
+        for pretix_slug in set(api_events).intersection(set(events)):
+            events[pretix_slug].update(api_events[pretix_slug])
+        for pretix_slug in set(events) - set(api_events):
+            events[pretix_slug].delete()
+    print(events)
```

Comment:
this `print` can go?

Reply:
let's discuss some sort of logging then! as a user (and developer), i don't like being totally in the dark what's happening.. :)

Reply:
we discussed. i removed this print, but adding some logging whenever we create/update/delete an event.